docs/concepts/tools/kubectl/object-management-using-declarative-config.md

```diff
@@ -0,0 +1,517 @@
+| No                                  | -                                  | No                           | Do nothing / Keep live value     |
+
+### Merging changes for list fields
+
+Merging changes to lists uses one of 2 strategies.  The strategy is controlled
+per-field on the Kubernetes object definition.
+
+- Replace: Treat the list the same as a primitive field.  Replace or delete the entire list.  Preserves ordering.
+- Merge: Treat the list like a map.  Add, delete, or update individual entries.  Does not preserve ordering.
+
+The merge strategy requires that a `mergeKey` be defined on the list field.
+The `mergeKey` is a field on the list elements to be used as the map key when
+merging the list as though it were a map.
+
+**Note:** As of Kubernetes 1.5, lists of primitive types do not support the merge strategy.
+
+{% comment %}
+TODO(1.6): Update this with the following for 1.6
+
+Lists of primitive types with the merge strategy use the value as the `mergeKey`.
+{% endcomment %}
+
+**Note:** As of Kubernetes 1.5, the strategy and `mergeKey`s for list
+fields are defined in the [types.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go)
+source code in the tags `patchStrategy` and `patchMergeKey`.
+
+{% comment %}
+TODO(pwittrock): Update this once the patchStrategy and patchMergeKey are
+represented in the open-api schema and shown in the reference docs.
+{% endcomment %}
+
+{% comment %}
+TODO(pwittrock): We need to support a flag to auto generate these labels + selectors for users.
+{% endcomment %}
+
+## Example map merge
+
+`last-applied-config` value - stored as an annotation on the live object
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+...
+spec:
+  replicas: 3
+  minReadySeconds: 10
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.10
+```
+
+Local Object Configuration
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+...
+spec:
+  replicas: 4
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.10
+```
+
+Live object before `apply`
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+...
+spec:
+  replicas: 3              # Will be changed to 4
+  minReadySeconds: 10      # Will be cleared
+  revisionHistoryLimit: 3  # Will be ignored
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.10
+```
+
+Live object after `apply`
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+...
+spec:
+  replicas: 4
+  revisionHistoryLimit: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.10
+```
+
+## Example list with merge strategy
+
+**Note:** This examples shows ordering in lists as being preserved so
```

Comment:
I will update the comment.  My intention was to say that the ordering WILL NOT be preserved in practice, even though the example keeps the same ordering.  This is done to make it easy to understand which elements are being added / deleted.